src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -136,6 +139,28 @@ object LazyModule
     if (!bc.suggestedNameVar.isDefined) bc.suggestName(valName.name)
     bc
   }
+
+  import chisel3.{Clock, Reset}
+
+  def withClock[T <: LazyModule](clock: => Clock)(bc: T)(implicit valName: ValName, sourceInfo: SourceInfo): T = {
+    val res = apply(bc)
+    res.wrap = (x: LazyModuleImpLike) => chisel3.withClock(clock)(x)
+    res
+  }
+
+  def withReset[T <: LazyModule](reset: => Reset)(bc: T)(implicit valName: ValName, sourceInfo: SourceInfo): T = {
+    val res = apply(bc)
+    res.wrap = (x: LazyModuleImpLike) => chisel3.withReset(reset)(x)
+    res
+  }
+
+  def withAndClock[T <: LazyModule](clock: => Clock, reset: => Reset)
```

Comment:
Did you intend this to be def withClockAndReset ... ?

Reply:
I did indeed! Whoops